tracer/src/Datadog.Trace/FeatureFlags/Exposure/ExposureApi.cs

```diff
@@ -0,0 +1,163 @@
+                { "env", settings.Environment ?? "unknown" },
+                { "version", settings.ServiceVersion ?? "unknown" }
+            };
+            Interlocked.Exchange(ref _context, context);
+        }
+
+        public static ExposureApi Create(TracerSettings tracerSettings)
+        {
+            return new ExposureApi(tracerSettings);
+        }
+
+        public void Dispose()
+        {
+            try { _processExit.TrySetResult(true); }
+            catch { }
+        }
+
+        public void TryToStartSendLoopIfNotStarted()
+        {
+            if (_started != 0 || Interlocked.Exchange(ref _started, 1) != 0)
+            {
+                return;
+            }
+
+            _ = Task.Run(SendLoopAsync).ContinueWith(t => { Log.Error(t.Exception, "FeatureFlags Exposure send loop failed"); }, TaskContinuationOptions.OnlyOnFaulted);
+        }
+
+        private async Task SendLoopAsync()
+        {
+            Log.Debug("ExposureApi::SendLoopAsync -> Enter");
+            while (!_processExit.Task.IsCompleted)
+            {
+                try
+                {
+                    var apiRequestFactory = _apiRequestFactory;
+                    if (apiRequestFactory is not null)
+                    {
+                        var uri = apiRequestFactory.GetEndpoint($"evp_proxy/v2/{ExposurePath}");
+                        if (_exposures.Count > 0)
+                        {
+                            await SendBatchAsync(apiRequestFactory, uri).ConfigureAwait(false);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Error while sending Feature Flags exposures to the agent");
+                }
+
+                try
+                {
+                    await Task.WhenAny(_processExit.Task, Task.Delay(_sendInterval)).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    // We are shutting down, so don't do anything about it
+                }
+
+                Log.Debug("ExposureApi::SendLoopAsync -> Exit");
+            }
+        }
+
+        private async Task SendBatchAsync(IApiRequestFactory apiRequestFactory, Uri uri)
+        {
+            try
+            {
+                var request = apiRequestFactory.Create(uri);
+                var payload = GetPayload();
+                using var response = await request.PostAsync(payload, MimeTypes.Json)
+                                        .ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error in Feature Flags exposures reporting loop");
+            }
+        }
+
+        private ArraySegment<byte> GetPayload()
+        {
+            ExposuresRequest request;
+            lock (_exposures)
+            {
+                var exposures = _exposures.ToList();
```

Comment:
I _think_ this can potentially give perf benefits in some TFMs
```suggestion
                List<ExposureEvent> exposures = [.._exposures];
```